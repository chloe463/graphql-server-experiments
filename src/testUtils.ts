import { ApolloServer } from "apollo-server";
import { Context } from "./context";
import { schema } from "./graphql/schema";

type DummyContext = {
  [key in keyof Context]: jest.Mock;
};

export const constructTestServer = (context: DummyContext) => {
  const server = new ApolloServer({
    schema,
    context,
  });

  return server;
};
