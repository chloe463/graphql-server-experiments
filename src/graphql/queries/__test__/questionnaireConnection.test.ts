import { describe, expect, it, jest } from "bun:test";
import { PrismaClient } from "@prisma/client";
import gql from "graphql-tag";
import { constructTestServer } from "../../../testUtils";

const prismaClientMock = new PrismaClient();

const GET_QUESTIONNAIRE_CONNECTION = gql`
  query GetQuestionnaireConnection($first: Int, $after: String) {
    questionnaireConnection(first: $first, after: $after) {
      totalCount
      pageInfo {
        hasNextPage
        hasPreviousPage
        startCursor
        endCursor
      }
      edges {
        node {
          id
          title
          description
          state
          startAt
          endAt
        }
        cursor
      }
    }
  }
`;

describe("[Query] questionnaireConnection", () => {
  it("returns questionnaire list", async () => {
    (prismaClientMock as any).questionnaire = {
      findMany: jest.fn().mockReturnValue([
        {
          id: 1,
          title: "Questionnaire.1",
          description: "ab accusantium ipsam voluptatibus vero voluptatum beatae et sed eum harum eveniet et fugiat deleniti laborum dolor ea perferendis ea eligendi molestias possimus molestiae quae et aspernatur et aut omnis",
          state: 1,
          startAt: new Date("2021-04-29T16:23:20.874Z"),
          endAt: new Date("2021-06-29T15:00:00.000Z"),
        },
        {
          id: 2,
          title: "Questionnaire.2",
          description: "voluptatibus magni et nesciunt quo commodi beatae ea error qui fugit minus nostrum dolores eaque quidem beatae error necessitatibus consequatur quibusdam sunt molestiae aut nulla dolor ad alias neque corrupti",
          state: 2,
          startAt: new Date("2021-04-29T16:23:20.874Z"),
          endAt: new Date("2021-06-29T15:00:00.000Z"),
        },
        {
          id: 3,
          title: "Questionnaire.3",
          description: "optio maiores similique dolorem aut voluptatem autem voluptatem ipsum exercitationem nihil voluptatem nemo illo quis in ut rerum et ad esse ipsa quae esse voluptas perspiciatis aspernatur possimus ut dolorem",
          state: 3,
          startAt: new Date("2021-04-29T16:23:20.874Z"),
          endAt: new Date("2021-06-29T15:00:00.000Z"),
        },
      ]),
      aggregate: jest.fn().mockReturnValue({
        _count: 3,
        _min: { id: 1 },
        _max: { id: 3 },
      }),
    };
    const server = constructTestServer();

    const res = await server.executeOperation(
      { query: GET_QUESTIONNAIRE_CONNECTION, variables: { first: 3, after: "0" } },
      {
        contextValue: {
          prismaClient: prismaClientMock,
        }
      }
    );
    expect(res).toMatchSnapshot();
  });
});
