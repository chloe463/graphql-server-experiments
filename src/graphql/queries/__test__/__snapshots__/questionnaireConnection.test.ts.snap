// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Query] questionnaireConnection returns questionnaire list 1`] = `
Object {
  "data": Object {
    "questionnaireConnection": Object {
      "edges": Array [
        Object {
          "cursor": "1",
          "node": Object {
            "description": "ab accusantium ipsam voluptatibus vero voluptatum beatae et sed eum harum eveniet et fugiat deleniti laborum dolor ea perferendis ea eligendi molestias possimus molestiae quae et aspernatur et aut omnis",
            "endAt": "2021-06-29T15:00:00.000Z",
            "id": 1,
            "startAt": "2021-04-29T16:23:20.874Z",
            "state": 1,
            "title": "Questionnaire.1",
          },
        },
        Object {
          "cursor": "2",
          "node": Object {
            "description": "voluptatibus magni et nesciunt quo commodi beatae ea error qui fugit minus nostrum dolores eaque quidem beatae error necessitatibus consequatur quibusdam sunt molestiae aut nulla dolor ad alias neque corrupti",
            "endAt": "2021-06-29T15:00:00.000Z",
            "id": 2,
            "startAt": "2021-04-29T16:23:20.874Z",
            "state": 2,
            "title": "Questionnaire.2",
          },
        },
        Object {
          "cursor": "3",
          "node": Object {
            "description": "optio maiores similique dolorem aut voluptatem autem voluptatem ipsum exercitationem nihil voluptatem nemo illo quis in ut rerum et ad esse ipsa quae esse voluptas perspiciatis aspernatur possimus ut dolorem",
            "endAt": "2021-06-29T15:00:00.000Z",
            "id": 3,
            "startAt": "2021-04-29T16:23:20.874Z",
            "state": 3,
            "title": "Questionnaire.3",
          },
        },
      ],
      "pageInfo": Object {
        "endCursor": "3",
        "hasNextPage": false,
        "hasPreviousPage": false,
        "startCursor": "1",
      },
      "totalCount": 3,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
